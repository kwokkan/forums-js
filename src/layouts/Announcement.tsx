import React from "react";
import { Message } from "rsuite";

interface IProps {
    title: string;
    message: string;
}

export function Announcement(props: IProps) {
    return (
        <Message
            showIcon
            type="info"
            title={props.title}
            description={<p>{props.message}</p>}
        >
        </Message>
    );
}
