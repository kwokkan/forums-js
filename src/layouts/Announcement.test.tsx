import React from "react";
import { create } from "react-test-renderer";
import { Announcement } from "./Announcement";

test("Renders with message", () => {
    const tree = create(
        <Announcement title="Testing" message="Rendered in test." />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
