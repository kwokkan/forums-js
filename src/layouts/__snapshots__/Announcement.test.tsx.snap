// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with message 1`] = `
<div
  className="rs-message rs-message-info rs-message-show rs-message-has-icon"
  role="alert"
  title="Testing"
>
  <div
    className="rs-message-container"
  >
    <div
      className="rs-message-icon-wrapper"
    >
      <svg
        aria-hidden={true}
        aria-label="info"
        className="rs-icon"
        data-category="legacy"
        fill="currentColor"
        focusable={false}
        height="1em"
        viewBox="0 0 14 14"
        width="1em"
      >
        <path
          d="M7 14A7 7 0 117 0a7 7 0 010 14zm0-9.333a1.167 1.167 0 100-2.334 1.167 1.167 0 000 2.334zm0 1.166c-.644 0-1.167.522-1.167 1.167v3.5a1.167 1.167 0 002.334 0V7c0-.644-.522-1.167-1.167-1.167z"
        />
      </svg>
    </div>
    <div
      className="rs-message-content"
    >
      <div
        className="rs-message-body"
      >
        <p>
          Rendered in test.
        </p>
      </div>
    </div>
  </div>
</div>
`;
